docs_src/microservices/device/remote/Ch-RemoteSecure.md

```diff
@@ -1,2 +1,342 @@
+   repository:
+
+     ```bash
+     git clone https://github.com/edgexfoundry/edgex-compose.git
+     ```
+
+3. Checkout the required version:
+
+      ```bash
+      git checkout {{edgexversion}}
+      ```
+
+### Generate SSH key pair for SSH tunnel
+1. Navigate to `edgex-examples/security/remote_devices/spiffe_and_ssh`
+2. Run `generate_keys.sh` script
+3. Copy `remote` folder located at `edgex-examples/security/remote_devices/spiffe_and_ssh` to remote nodes
+
+### Edit docker-compose file
+1. Navigate to `edgex-examples/security/remote_devices/spiffe_and_ssh/local` and edit
+   `docker-compose.yml` file
+ 
+    !!! note
+        Below instructions explains to run 2 instances of usb service: instance 1 and instance 2)
+
+2. Change `host_ip` to ip address of local system for `consul`, `core-metadata`, `database` and `vault`
+
+```yaml
+services:
+  consul:
+    ports:
+      - mode: ingress
+        host_ip: 192.168.4.103    # insert local machine's host ip here
+        target: 8500
+        published: "8500"
+        protocol: tcp
+
+  core-metadata:
+    ports:
+      - mode: ingress
+        host_ip: 192.168.4.103    # insert local machine's host ip here
+        target: 59881
+        published: "59881"
+        protocol: tcp
+  database:
+    ports:
+      - mode: ingress
+        host_ip: 192.168.4.103    # insert local machine's host ip here
+        target: 6379
+        published: "6379"
+        protocol: tcp
+  vault:
+    ports:
+      - mode: ingress
+        host_ip: 192.168.4.103    # insert local machine's host ip here
+        target: 8200
+        published: "8200"
+        protocol: tcp
+```
+1. Add/edit environment variables for `consul`, `security-proxy-setup` and `security-secretstore-setup` services
+
+```yaml
+  services:
+    consul:
+      environment:
+        EDGEX_ADD_REGISTRY_ACL_ROLES: "device-usb-camera_1,device-usb-camera_2"
+
+    security-proxy-setup:
+      environment:
+        EDGEX_ADD_PROXY_ROUTE: "device-usb-camera_1.http://192.168.4.104:59983,device-usb-camera_2.http://192.168.4.105:59983"
+        ROUTES_DEVICE_USB_CAMERAL_HOST: "device-usb-camera_1,device-usb-camera_2"
+
+    security-secretstore-setup:
+      environment:
+        EDGEX_ADD_KNOWN_SECRETS: "redisdb[app-rules-engine],redisdb[device-usb-camera_1],message-bus[device-usb-camera_1],redisdb[device-usb-camera_2],message-bus[device-usb-camera_2]"
+```
+
+1. Add multiple `device-ssh-proxy` service and configure with remote node ips
+   - copy `device-ssh-proxy` service and paste it again below itself.
+   - rename services to `device-ssh-proxy-1` and `device-ssh-proxy-2` respectively
+   - Modify the existing environment variables and other fields for `device-ssh-proxy-1`
+   
+   ```yaml
+     services:
+       device-ssh-proxy-1:
+         container_name: edgex-device-ssh-proxy-1
+         environment:
+           SERVICE_HOST: edgex-device-usb-camera
+           SERVICE_PORT: 59983
```

Comment:
I do not believe so, as the steps show port mapping of 59984 external to 59983 internal. I'm not sure if it would make things simpler if the internal and external were both changed.

Reply:
Right that is changing the internal.